src/provider/file/mod.rs

```diff
@@ -151,13 +157,24 @@ impl<T: Search + Send + Sync> FileProvider<T> {
 #[async_trait::async_trait]
 impl<T: crate::search::Search + Send + Sync> Provider for FileProvider<T> {
     #[instrument(level = "trace", skip(self, inv), fields(id = %inv.bindle.id))]
-    async fn create_invoice(&self, inv: &crate::Invoice) -> Result<Vec<crate::Label>> {
+    async fn create_invoice(
+        &self,
+        inv: &mut crate::Invoice,
+        role: SignatureRole,
+        secret_key: &SecretKeyEntry,
+        strategy: VerificationStrategy,
```

Comment:
Nit: Does it make sense for `create_invoice` to own `strategy` or can we just do a borrow? Looks like we could just do a borrow, but maybe that doesn't matter too much

Reply:
I'm not sure why I would prefer a borrow in this case. Can you explain why that would be better?

Reply:
I think of it from a consumer point of view. If I have a verification strategy I created, now I have to clone it if I plan on using it anywhere else when I call this function. As far as I could tell, this function doesn't have any need for ownership (it isn't storing it somewhere, and is just using it to access data). With something like this very lightweight enum, it doesn't _truly_ matter, hence the nit. But in general I like to think of the question "Do I need to take ownership from the user?"

Reply:
That makes sense